src/commonMain/kotlin/org/kosat/heuristics/VariableSelector.kt

```diff
@@ -16,38 +16,61 @@ abstract class VariableSelector {
     abstract fun nextDecisionVariable(vars: List<CDCL.VarState>, level: Int): Int
     abstract fun addVariable()
     abstract fun update(lemma: Clause)
+    abstract fun backTrack(variable: Int)
 }
 
-class VSIDS(private var varsNumber: Int = 0) : VariableSelector() {
+class VSIDS(private var varsNumber: Int = 0, private val vars: MutableList<CDCL.VarState>) : VariableSelector() {
     private val decay = 50
-    private val divisionCoeff = 2.0
+    private val multiplier = 2.0
     private var numberOfConflicts = 0
+    private var scoreInc = 1.0
+    private var incLimit = 1e100
+
+    // list of scores for variables
+    private val scores = mutableListOf<Double>()
```

Comment:
I thought it is always called "activity". Don't know whether changing it to "scores" makes it better.